File: AudioPlay/pre/EOCAudioClassOne/EOCAudioClassOne/ViewController.m
//
//  ViewController.m
//  EOCAudioClassOne
//
//  Created by EOC on 2017/6/28.
//  Copyright © 2017年 EOC. All rights reserved.
//

#import "ViewController.h"
#import "EOCAudioPlayerViewCtr.h"
#import "AudioPlayerViewCtr.h"

@interface ViewController ()

@end

@implementation ViewController

- (void)viewDidLoad {
    
    [super viewDidLoad];

    [self testAudioStream];
}

- (void)eocAudioStream{
    
    EOCAudioPlayerViewCtr *eocAudioPlayerViewCtr = [EOCAudioPlayerViewCtr new];
    
    [self.view addSubview:eocAudioPlayerViewCtr.view];
    [self addChildViewController:eocAudioPlayerViewCtr];

}

- (void)testAudioStream{
    
    AudioPlayerViewCtr *audioPlayVCtr = [[AudioPlayerViewCtr alloc] init];
    [self.view addSubview:audioPlayVCtr.view];
    audioPlayVCtr.view.frame = ({
        CGRect rect = audioPlayVCtr.view.frame;
        rect.origin.y = 100;
        rect;
    });
    
    [self addChildViewController:audioPlayVCtr];
    
}







@end

File: CoreTextClass/CoreTextClass/ViewController.swift
//
//  ViewController.swift
//  CoreTextClass
//
//  Created by Casey on 30/01/2019.
//  Copyright © 2019 n. All rights reserved.
//

import UIKit

//https://reactnative.cn/docs/state/
class ViewController: UIViewController {

    override func viewDidLoad() {
        super.viewDidLoad()
        
        self.view.backgroundColor = .white
        
        
        let textLabel =  ImageTextLabel.init(frame: CGRect.init(x: 10, y: 100, width: 200, height: 200))
        self.view.addSubview(textLabel)
        
        textLabel.backgroundColor = .yellow
        
        
        let textVLabel =  VerticalGlyphLabel.init(frame: CGRect.init(x: 10, y: 300, width: 200, height: 200))
        self.view.addSubview(textVLabel)
        textVLabel.backgroundColor = .red
        
    }
    
    
    


}

